src/Runner.Common/RunServer.cs

```diff
@@ -64,5 +67,18 @@ public Task CompleteJobAsync(Guid planId, Guid jobId, TaskResult result, Diction
             return RetryRequest(
                 async () => await _runServiceHttpClient.CompleteJobAsync(requestUri, planId, jobId, result, outputs, stepResults, cancellationToken), cancellationToken);
         }
+
+        public Task<RenewJobResponse> RenewJobAsync(Guid planId, Guid jobId, CancellationToken cancellationToken)
+        {
+            CheckConnection();
+            var renewJobResponse = RetryRequest<RenewJobResponse>(
+                async () => await _runServiceHttpClient.RenewJobAsync(requestUri, planId, jobId, cancellationToken), cancellationToken);
+            if (renewJobResponse == null)
+            {
+                throw new TaskOrchestrationJobNotFoundException(jobId.ToString());
```

Comment:
it's a little bit weird that an empty response means not found, should we get something like 404 from the service and wrap it with some exception under `runServiceHttpClient`?

Reply:
Hm, need to thnk about it, creating more exceptions seems bit wasteful, but it's part of handling non vssf connections that affects results client and all other clients, I captured your point at https://github.com/github/actions-service-team/issues/3297, we will address these as part of that bundle, hoping that's okay!